### DISCUSSION TITLE
Display the translated name of the parent record

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#6047] [Help] [answered]
How can I display the translated name of the parent record?
![7](https://user-images.githubusercontent.com/29732308/228379744-dddec2da-3d63-4ae5-840b-eb1a5fe9cbb0.png)

**Comment by danharrin** [comment, marked as answer]
`formatStateUsing(fn ($state) => __($state))`

**Reply by ArtMin96** [reply]
Thanks, Dan. I don't mean that. I have added [translatable](https://filamentphp.com/plugins/spatie-translatable) to the filament demo category and I am trying to show the translated name of the parent.

```php
TextColumn::make('parent.name')
    ->label('Parent')
    ->searchable()
    ->sortable(),
```

**Reply by danharrin** [reply]
Same thing but not `__()`?

**Reply by ArtMin96** [reply]
`name` column:
```json
{"en":"Category en","hy":"Category hy","ru":"Category ru"}
```

In Spatie translatable package we can get it like this, but in Filament table I don't know how to get the same
```php
$item->getTranslation('name', 'en');
```

**Reply by danharrin** [reply]
`getStateUsing(fn ($record) => $record->parent->getTranslation('name', 'en'))`

**Reply by ArtMin96** [reply]
Thank you. This is what I needed. I don't know how I can mark your last comment as answered :grinning:
